sgt-back-end/server.js

```diff
@@ -0,0 +1,147 @@
+import express from 'express';
+import pg from 'pg';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', async (req, res) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      res.status(400).json({ error: '"gradeId" must be a positive integer' });
+      return;
+    }
+
+    const sql = `
+      select *
+        from "grades"
+      where "gradeId" = $1
+    `;
+
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (grade) {
+      res.json(grade);
+    } else {
+      res
+        .status(404)
+        .json({ error: `Cannot find grade with gradeId ${gradeId}` });
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occurred.' });
+  }
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    const grade = result.rows;
+    res.status(200).json(grade);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'An unexpected error occured.' });
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    if (!name || !course || !score) {
+      res.status(400).json({ error: 'The grade is invalid.' });
```

Comment:
You have to return here or you'll continue executing the code.

You have this mistake in all your endpoints. Fix them all.